tapfreighter/chain_porter.go

```diff
@@ -116,31 +116,39 @@ func (p *ChainPorter) Start() error {
 	p.startOnce.Do(func() {
 		log.Infof("Starting ChainPorter")
 
-		// Before we re-launch the main goroutine, we'll make sure to
-		// restart any other incomplete sends that may or may not have
-		// had the transaction broadcaster.
+		// Start the main chain porter goroutine.
+		p.Wg.Add(1)
+		go p.assetsPorter()
+
+		// Identify any pending parcels that need to be resumed and add
+		// them to the exportReqs channel so they can be processed by
+		// the main porter goroutine.
 		ctx, cancel := p.WithCtxQuit()
 		defer cancel()
-		pendingParcels, err := p.cfg.ExportLog.PendingParcels(ctx)
+		outboundParcels, err := p.cfg.ExportLog.PendingParcels(ctx)
 		if err != nil {
 			startErr = err
 			return
 		}
 
+		// Convert the outbound parcels into pending parcels so we can
+		// resume delivery using the normal parcel delivery mechanism.
+		pendingParcels := make(
+			[]*PendingParcel, 0, len(outboundParcels),
+		)
+		for _, outboundPkg := range outboundParcels {
+			pendingParcel := NewPendingParcel(outboundPkg)
+			pendingParcels = append(pendingParcels, pendingParcel)
+		}
+
 		log.Infof("Resuming delivery of %v pending asset parcels",
 			len(pendingParcels))
 
-		// Now that we have the set of pending sends, we'll make a new
-		// goroutine that'll drive the state machine till the broadcast
-		// point (which we might be repeating), and final terminal
-		// state.
+		// At this point the asset porter should be running, so it will
+		// pick up the pending parcels and attempt to deliver them.
 		for _, parcel := range pendingParcels {
-			p.Wg.Add(1)
-			go p.resumePendingParcel(parcel)
+			p.exportReqs <- parcel
```

Comment:
Why not just:
```
		for idx := range outboundParcels {
			p.exportReqs <- NewPendingParcel(outboundParcels[idx])
		}
```

Then we don't need the loop above. And fix a potential loop variable issue.

Reply:
Also, should we add the log message back here?
```
	log.Infof("Attempting to resume delivery for anchor_txid=%v",
		pkg.AnchorTx.TxHash().String())
```